src/MediaWiki/Hooks/UserChange.php

```diff
@@ -55,15 +55,15 @@ public function setOrigin( $origin ) {
 	/**
 	 * @since 3.0
 	 *
-	 * @param User|string $user
+	 * @param UserIdentity|string $user
 	 */
 	public function process( $user ) {
 
 		if ( !$this->namespaceExaminer->isSemanticEnabled( NS_USER ) ) {
 			return false;
 		}
 
-		if ( $user instanceof User ) {
+		if ( $user instanceof UserIdentity ) {
```

Comment:
This will fail for older versions of MediaWiki that don't have UserIdentity, I think. It probably needs to be
`( $user instanceof User || $user instanceof UserIdentity )`

Reply:
UserIdentity is @since 1.31. Is SemanticMediaWiki supposed to support even older versions?

Reply:
MW dev-master supports MW 1.35+

Reply:
I checked this before merging. Indeed since MW 31. Not an issue